packages/graphile-utils/src/makeChangeNullabilityPlugin.ts

```diff
@@ -1,49 +1,121 @@
 import type {
-  SchemaBuilder,
   Build,
   Context,
-  Plugin,
   Options,
+  Plugin,
+  SchemaBuilder,
 } from "graphile-build";
-import type { GraphQLInputFieldConfig, GraphQLFieldConfig } from "graphql";
+import {
+  GraphQLFieldConfig,
+  GraphQLInputFieldConfig,
+  GraphQLInputType,
+  GraphQLNonNull,
+  GraphQLType,
+} from "graphql";
+import { GraphQLList } from "graphql/type/definition";
+
+interface NullabilityOpts {
+  type?: boolean | boolean[];
+  args?: {
+    [argName: string]: boolean | boolean[];
+  };
+}
 
 interface ChangeNullabilityRules {
   [typeName: string]: {
-    [fieldName: string]: boolean;
+    [fieldName: string]: boolean | boolean[] | NullabilityOpts;
   };
 }
 
 export default function makeChangeNullabilityPlugin(
   rules: ChangeNullabilityRules
 ): Plugin {
   return (builder: SchemaBuilder, _options: Options) => {
+    function applyNullability<Type extends GraphQLType | GraphQLInputType>(
+      type: Type,
+      allNullabilities: boolean[]
+    ): Type {
```

Comment:
This is technically not true. E.g. we might feed in `GraphQLList<GraphQLObjectType>` but output `GraphQLList<GraphQLNonNull<GraphQLObjectType>>`.

I think you should remove the generic from this function.

Reply:
(And the other functions.)

Reply:
✅